### PR TITLE
[rllib] MAML Transform

### PR DESCRIPTION
**Pull request by michaelzhiluo**
Uses `transform()` method in Ray Iter API for MAML for Inner Adaptation Steps

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/latest/.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failure rates at https://ray-travis-tracker.herokuapp.com/.
- Testing Strategy
   - [x] Unit tests
   - [x] Release tests
   - [ ] This PR is not tested (please justify below)

### CODE REVIEW

File: rllib/agents/maml/maml.py [hunk 1]
```diff
@@ -79,30 +78,26 @@ def __init__(self, workers, inner_adaptation_steps, metric_gen):
...
-            # Reporting Adaptation Rew Diff
-            ep_rew_pre = self.metrics["episode_reward_mean"]
-            ep_rew_post = self.metrics["episode_reward_mean_adapt_" +
-                                       str(self.n)]
-            self.metrics["adaptation_delta"] = ep_rew_post - ep_rew_pre
-            return [(out, self.metrics)]
-        else:
-            self.inner_adaptation_step(samples)
-            return []
+    def __call__(self, itr):
```
Comment (ericl) [inline rllib/agents/maml/maml.py:81]: Can you make this whole thing a plain function instead of a class now?

Comment (michaelzhiluo) [reply rllib/agents/maml/maml.py:81]: There is still some internal states that needs to be preserved; for example, the number of inner adaptation steps for MAML.

**Comment by ericl** [comment]
You can keep that state as local variables in the function.

**Comment by michaelzhiluo** [comment]
@ericl Correct me if I'm wrong, unless the transform is a nested function within the distributed execution plan method (`execution_plan()` in `maml.py`), you cannot pass in arguments to the transform method. (it only accepts ParallelIterator)

**Comment by ericl** [comment]
You can capture the arguments inside the function, if you define the function inside the execution_plan() function.

**Comment by michaelzhiluo** [comment]
Aight lmk if this looks to you @ericl

**Review by ericl — APPROVED** [review]

**Pull request merged**